src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
That inlining shouldn't be possible (or at least, the inlining should successfully replicate the non-inlined equivalent). If inlining caused a bug I'd say the bug was with the compiler rather than the assembly (as it is, the behaviour I mentioned here seems like a compiler bug to me) but having something that would flag in CI if such an over-reaching inline did have might be a reasonable precaution.